SparkPost/support-docs
# WE-209: convert config files to ts

### What Changed
This PR brings our gatsby instance into alignment with our all complete AC of building this project in type script.

- It turns on type checking by default in all directories
- Creates tsx versions of our ts-config files and exports them to the root in js (as gatsby expects. it's unfortunately the only way gatsby will truly work with typescript)
- This also updates a lot of the node packages to their latest versions at this time.

### How to test
- just in case, delete your node_modules directory and run `npm install`. 
  - There are minor upgrades to some of these that mdx and matchbox will throw a fit over sometimes, so you might need to run `npm install --force` (it should still work since it worked locally for me, but verify as part of the test)
- verify that type checking is being enforced in the template file and any other files you want to check.
